lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -3079,6 +3110,56 @@ bool ExprAvailabilityWalker::diagnoseDeclRefAvailability(
   return false;
 }
 
+/// Diagnose uses of API annotated '@unavailableFromAsync' when used from
+/// asynchronous contexts.
+/// Returns true if a diagnostic was emitted, false otherwise.
+static bool
+diagnoseDeclUnavailableFromAsync(const ValueDecl *D, SourceRange R,
+                                 const Expr *call, const ExportContext &Where) {
+  // FIXME: I don't think this is right, but I don't understand the issue well
```

Comment:
This is a bit of a depressing comment, but... okay. There's certainly a bug somewhere if we're getting this code before constraint application.

Reply:
Yeah, I'm probably going to need some help with someone more familiar with the constraint solver and result builders to get this fully debugged.